templates/commands/render.go

```diff
@@ -322,6 +343,92 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// processUnknownInputs checks for any unknown input flags and returns an error.
+func processUnknownInputs(flagInputs map[string]string, spec *model.Spec) error {
+	var merr error
+
+	specInputs := make(map[string]any)
+	for _, v := range spec.Inputs {
+		specInputs[v.Name.Val] = struct{}{}
+	}
+
+	for key, value := range flagInputs {
+		if _, ok := specInputs[key]; !ok {
+			merr = errors.Join(merr, fmt.Errorf("  - %s=%s", key, value))
```

Comment:
The convention that we use in spec.go is that each error is fully specified, not just a bullet point. `an input with an unknown key of %q was provided, that is not an input accepted by this template` or something.